hyp3lib/get_dem.py

```diff
@@ -15,16 +15,70 @@
 from osgeo import ogr
 from osgeo import osr
 from pyproj import Transformer
+import configparser
+from hyp3lib.asf_geometry import raster_meta, geotiff2data, data2geotiff, geometry2shape
+from hyp3lib.fetch import download_file
 
+'''
 import hyp3lib.etc
 from hyp3lib import DemError
 from hyp3lib import dem2isce
 from hyp3lib import saa_func_lib as saa
 from hyp3lib.asf_geometry import raster_meta
 from hyp3lib.fetch import download_file
+'''
+
+
+def vector_meta(vectorFile):
+
+    vector = ogr.Open(vectorFile)
+    layer = vector.GetLayer()
+    layerDefinition = layer.GetLayerDefn()
+    fieldCount = layerDefinition.GetFieldCount()
+    fields = []
+    for ii in range(fieldCount):
+        field = {}
+        field['name'] = layerDefinition.GetFieldDefn(ii).GetName()
+        field['type'] = layerDefinition.GetFieldDefn(ii).GetType()
+        field['width'] = layerDefinition.GetFieldDefn(ii).GetWidth()
+        field['precision'] = layerDefinition.GetFieldDefn(ii).GetPrecision()
+        fields.append(field)
+    proj = layer.GetSpatialRef()
+    extent = layer.GetExtent()
+    features = []
+    featureCount = layer.GetFeatureCount()
+    for kk in range(featureCount):
+        value = {}
+        feature = layer.GetFeature(kk)
+        for ii in range(fieldCount):
+            if fields[ii]['type'] == ogr.OFTInteger:
+                value[fields[ii]['name']] = int(feature.GetField(ii))
+            elif fields[ii]['type'] == ogr.OFTReal:
+                value[fields[ii]['name']] = float(feature.GetField(ii))
+            else:
+                value[fields[ii]['name']] = feature.GetField(ii)
+        value['geometry'] = feature.GetGeometryRef().ExportToWkt()
+        features.append(value)
+
+    return (fields, proj, extent, features)
+
+
+def readFileList(configFile, section):
+
+    files = []
+    fileList = False
+    lines = [line.rstrip('\n') for line in open(configFile)]
```

Comment:
This leaves the file open to be garbage collected instead of explicitly closing.